presto-main/src/main/java/com/facebook/presto/sql/gen/PreGeneratedExpressions.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql.gen;
+
+import com.facebook.presto.bytecode.MethodDefinition;
+import com.facebook.presto.sql.relational.CallExpression;
+
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class PreGeneratedExpressions
```

Comment:
I think differently on this topic.

The implementation of lambda expression is a piece of code that invokes the MethodHandle stored as a static final field corresponding to a method pre-generated for the lambda expression. Implementation of try expression is similar. That's why I don't like "ExpressionImplementations", "CompiledExpressions" or something along those lines.

Since the introduction of try, the generation of byte code for expression is two steps: a prepare phase and a main phase. 

This class is introduced to avoid passing multiple maps around between the prepare phase and the main phase. It helps encapsulate those maps into an entity. It serves a very specific function. And the current name explains that specific function.

If we ever need to generalize the function of this class, we can rename it at that point. We will be able to come up with a name appropriate to the function this class serves at that point. A few things around this class will likely be re-organized as well. Names that are not appropriate right now might be appropriate under the new context.

For the above reasons combined, I prefer using `PreGeneratedExpressions` as the name of this class.